backend/pkg/api/handle_kafka_connect_integration_test.go

```diff
@@ -0,0 +1,399 @@
+	"github.com/docker/go-connections/nat"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"github.com/twmb/franz-go/pkg/kgo"
+	"go.uber.org/zap"
+
+	"github.com/redpanda-data/console/backend/pkg/config"
+	"github.com/redpanda-data/console/backend/pkg/connect"
+)
+
+func (s *APIIntegrationTestSuite) TestHandleCreateConnector() {
+	t := s.T()
+
+	require := require.New(t)
+	assert := assert.New(t)
+
+	// setup
+	ctx := context.Background()
+
+	connectTestNetwork := "api_integration_redpanda_connect_test_network"
+
+	// create one common network that all containers will share
+	testNetwork, err := testcontainers.GenericNetwork(ctx, testcontainers.GenericNetworkRequest{
+		ProviderType: testcontainers.ProviderDocker,
+		NetworkRequest: testcontainers.NetworkRequest{
+			Name:           connectTestNetwork,
+			CheckDuplicate: true,
+			Attachable:     true,
+		},
+	})
+	require.NoError(err)
+
+	commonNetwork := testNetwork
+
+	// Redpanda container
+	exposedPlainKafkaPort := rand.Intn(50000) + 10000 //nolint:gosec // We can use weak random numbers for ports in tests.
+	exposedOutKafkaPort := rand.Intn(50000) + 10000   //nolint:gosec // We can use weak random numbers for ports in tests.
+	exposedKafkaAdminPort := rand.Intn(50000) + 10000 //nolint:gosec // We can use weak random numbers for ports in tests.
+
+	redpandaContainer, err := runRedpandaForConnect(ctx, connectTestNetwork, exposedPlainKafkaPort, exposedOutKafkaPort, exposedKafkaAdminPort)
+	require.NoError(err)
+
+	seedBroker, err := getMappedHostPort(ctx, redpandaContainer, nat.Port(strconv.FormatInt(int64(exposedOutKafkaPort), 10)+"/tcp"))
+	require.NoError(err)
+
+	// HTTPBin container
+	httpC, err := runHTTPBin(ctx, connectTestNetwork)
+	require.NoError(err)
+
+	httpBinContainer := httpC
+
+	// Kafka Connect container
+	connectC, err := runConnect(connectTestNetwork, []string{"redpanda:" + strconv.FormatInt(int64(exposedPlainKafkaPort), 10)})
+	require.NoError(err)
+
+	connectContainer := connectC
+
+	connectPort, err := connectContainer.MappedPort(ctx, nat.Port("8083"))
+	require.NoError(err)
+
+	connectHost, err := connectContainer.Host(ctx)
+	require.NoError(err)
+
+	// new connect service
+	log, err := zap.NewProduction()
+	require.NoError(err)
+
+	connectCfg := config.Connect{}
+	connectCfg.SetDefaults()
+	connectCfg.Enabled = true
+	connectCfg.Clusters = []config.ConnectCluster{
+		{
+			Name: "redpanda_connect",
+			URL:  "http://" + connectHost + ":" + connectPort.Port(),
+		},
+	}
+
+	newConnectSvc, err := connect.NewService(connectCfg, log)
+	assert.NoError(err)
+
+	// save old
+	oldConnectSvc := s.api.ConnectSvc
+
+	// switch
+	s.api.ConnectSvc = newConnectSvc
+
+	// allow for connect service to establish connection to connect cluster
+	timer := time.NewTimer(200 * time.Millisecond)
+	<-timer.C
```

Comment:
I have removed this timer since it's not needed as we do the connection check synchronously within the service. There is only one other timer in tests in this repo, and it's a timer to allow for connect instance to do some work and write data into the kafka topic.